app/controllers/openid_connect/logout_controller.rb

```diff
@@ -113,15 +115,25 @@ def handle_successful_logout_request(result, redirect_uri)
         @service_provider_name = @logout_form.service_provider&.friendly_name
         delete_branded_experience(logout: true)
 
-        render :index
+        render :confirm_logout
       else
-        analytics.logout_initiated(**result.to_h.except(:redirect_uri))
-        irs_attempts_api_tracker.logout_initiated(success: result.success?)
+        handle_logout(result, redirect_uri)
+      end
+    end
 
-        sign_out
+    def handle_logout(result, redirect_uri)
+      analytics.logout_initiated(**to_event(result))
+      irs_attempts_api_tracker.logout_initiated(success: result.success?)
 
-        redirect_user(redirect_uri, @logout_form.service_provider&.issuer, current_user&.uuid)
-      end
+      sign_out
+
+      redirect_user(redirect_uri, @logout_form.service_provider&.issuer, current_user&.uuid)
```

Comment:
Noting that the reordering `redirect_user` to happen after `sign_out` caused a change in behavior with `current_user` always being `nil`.  It wasn't caught in our specs since our `stub_sign_in` helper wasn't faithfully recreating the behavior of Devise's handling of `current_user`

I accidentally stumbled into this in #10887 with a change to improve the stubbed sign-out behavior. More info at https://github.com/18F/identity-idp/pull/10887/files#r1661091791.